app/serializers/api/tracks_serializer.rb

```diff
@@ -0,0 +1,98 @@
+class API::TracksSerializer
+  include Rails.application.routes.url_helpers
+  include Mandate
+
+  def initialize(tracks, user = nil)
+    @tracks = tracks
+    @user = user
+  end
+
+  def to_hash
```

Comment:
So the purpose of this is purely to serialize data for outputs. It's quite deliberately not a "thing" in its own right. It shouldn't have any behaviour etc, and just has methods for `to_json` and `to_hash` for when that data is serialized. It's really just an object that outputs a view of something.

I've modelled it as a serializer Object, but we could equally have it as a command (`SerializeTracks`) instead - I'd be quite happy with that.

Reply:
> I've modelled it as a serializer Object, but we could equally have it as a command (SerializeTracks) instead - I'd be quite happy with that.

I like that!

Reply:
OK, I can do that as a seperate PR.

Reply:
I'd argue it's actually a thing, but wouldn't want to press that right now. I'm happy with having this as a command to serialize tracks, but would argue in the future that it would make our code less flexible. Having explicit classes for commands is less flexible than running `to_hash` on the object you want to serialize.

Reply:
@kntsoriano Continued this discussion [here](https://github.com/exercism/v3-website/issues/115) as a stand-alone issue 🙂